pkg/apis/azureprovider/v1beta1/types.go

```diff
@@ -401,11 +401,16 @@ type Image struct {
 type VMIdentity string
 
 type OSDisk struct {
-	OSType      string      `json:"osType"`
-	ManagedDisk ManagedDisk `json:"managedDisk"`
-	DiskSizeGB  int32       `json:"diskSizeGB"`
+	OSType      string                `json:"osType"`
+	ManagedDisk ManagedDiskParameters `json:"managedDisk"`
+	DiskSizeGB  int32                 `json:"diskSizeGB"`
 }
 
-type ManagedDisk struct {
-	StorageAccountType string `json:"storageAccountType"`
+type ManagedDiskParameters struct {
+	StorageAccountType string                       `json:"storageAccountType"`
+	DiskEncryptionSet  *DiskEncryptionSetParameters `json:"diskEncryptionSet,omitempty"`
+}
+
+type DiskEncryptionSetParameters struct {
+	ID string `json:"id,omitempty"`
```

Comment:
I'd consider resourceID rather than id - may be a little clearer.

Reply:
also note elsewhere this API already uses resourceID...

Reply:
Again, aligning with what's in the upstream repository:
https://github.com/Azure/azure-sdk-for-go/blob/master/services/compute/mgmt/2019-12-01/compute/models.go#L2408

Reply:
Yeah, I like aligning with upstream.

Reply:
+1 to leave as-is